PostWork04

FranciscoCamas opened the pull request:
Buenas noches compañeros.
Termine el PostWork04.

Este PostWork cumple con lo que nos solicitan por BEDU:

1.-Utiliza el proyecto de **postwork 3.**
2.-Crea un nuevo package con el nombre de **async**.
3.-Genera el modelo **SolicitudEstudiante** el cual tendrá:
   a. Constructor para recibir Estudiante y Curso.
   b. Métodos get y set tanto para Estudiante como para Curso.
4.-Crea la interfaz **NotificadorInscripcion**, la cual notificará al maestro cada que se reciba una solicitud. 
5.-Genera la clase **ReceptorSolicitudes** “Debe implementar de Runnable” la cual se encargará de:
  a. Procesar y esperar las solicitudes dentro del run(). 
  b. Iniciar y detener la ejecución. “Cada uno en su método correspondiente”.
  c. Agregar nuevas solicitudes a la lista.
  d. Retornar si se encuentra en ejecución. “A través de un método”.
6.-Crea la clase **InscripcionAlumnos** “Contendrá el main” la cual se encargará de generar:
  a. Cursos (Por lo menos cuatro).
  b. Estudiantes (Por lo menos veinte).
7.-**InscripcionAlumnos** deberá agregar las solicitudes, con sus respectivos estudiantes y cursos. además notificar con un event Loop cuando:
  a. Un alumno se inscribe a un curso.
  b. En un **worker** al maestro de que se realizó dicha inscripción.

 Adjunto una imagen del clase corriendo y desplegando los mensajes para "notificar" al profesor.

 SI algún detalle se me olvido, porfa me comentan mañana.

 Saludos.

atte. Francisco.

FranciscoCamas committed:
PostWork04

Buenas noches compañeros.
Termine el PostWork04.

Este PostWork cumple con lo que nos solicitan por BEDU:

1.-Utiliza el proyecto de postwork 3.
2.-Crea un nuevo package con el nombre de async.
3.-Genera el modelo SolicitudEstudiante el cual tendrá:
   a. Constructor para recibir Estudiante y Curso.
   b. Métodos get y set tanto para Estudiante como para Curso.
4.-Crea la interfaz NotificadorInscripcion, la cual notificará al maestro cada que se reciba una solicitud.
5.-Genera la clase ReceptorSolicitudes “Debe implementar de Runnable” la cual se encargará de:
  a. Procesar y esperar las solicitudes dentro del run(). 
  b. Iniciar y detener la ejecución. “Cada uno en su método correspondiente”.
  c. Agregar nuevas solicitudes a la lista.
  d. Retornar si se encuentra en ejecución. “A través de un método”.
6.-Crea la clase InscripcionAlumnos “Contendrá el main” la cual se encargará de generar:
  a. Cursos (Por lo menos cuatro).
  b. Estudiantes (Por lo menos veinte).
7.-InscripcionAlumnos deberá agregar las solicitudes, con sus respectivos estudiantes y cursos. además notificar con un event Loop cuando:
  a. Un alumno se inscribe a un curso.
  b. En un worker al maestro de que se realizó dicha inscripción.

 Adjunto una imagen del clase corriendo y desplegando los mensajes para "notificar" al profesor.

 SI algún detalle se me olvido, porfa me comentan mañana.

 Saludos.

atte. Francisco.

merolhack commented:
No hay archivo pom.xml en el directorio pw4, no se pueden ejecutar las pruebas unitarias:

- https://github.com/merolhack/bedu-equipo5-m3-postworks/tree/FranciscoCamas-patch-2/pw4

<img width="1271" alt="imagen" src="https://user-images.githubusercontent.com/157202/236213673-e1844de6-49ba-4d9a-9ba8-63caeaf87f20.png">

FranciscoCamas committed:
POM Faltante

Buenos días Lennin.

 Una disculpa, no lo subí en el request anterior.

Saludos.

 atte. Francisco.

Commit:
Merge remote-tracking branch 'origin/FranciscoCamas-patch-3' into FranciscoCamas-patch-2

Commit:
Merge branch 'main' into FranciscoCamas-patch-2

Commit:
:sparkles: feat(pw4): Unit tests fixed

merolhack reviewed (approved):
Se ejecuta correctamente la clase "InscripcionAlumnos":
<img width="1136" alt="imagen" src="https://user-images.githubusercontent.com/157202/236310053-b21703c9-e663-4657-9ca4-75b6c70ba792.png">
